lib/api/errors/app-error.ts

```
@@ -0,0 +1,35 @@
+// lib/api/errors/app-error.ts
+
+/**
+ * Base error class for application errors
+ */
+export class AppError extends Error {
+  code: string;
+  status: number;
+  details?: Record<string, any>;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)